packages/actions/src/helpers/setup.ts

```diff
@@ -0,0 +1,176 @@
+        const wasmStorageFilePath = getWasmStorageFilePath(circuitPrefix, wasmCompleteFilename)
+        const potStorageFilePath = getPotStorageFilePath(smallestPowersOfTauCompleteFilenameForCircuit)
+        const zkeyStorageFilePath = getZkeyStorageFilePath(circuitPrefix, firstZkeyCompleteFilename)
+  
+        const files: any = {
+            potFilename: smallestPowersOfTauCompleteFilenameForCircuit,
+            r1csFilename: r1csCompleteFilename,
+            wasmFilename: wasmCompleteFilename,
+            initialZkeyFilename: firstZkeyCompleteFilename,
+            potStoragePath: potStorageFilePath,
+            r1csStoragePath: r1csStorageFilePath,
+            wasmStoragePath: wasmStorageFilePath,
+            initialZkeyStoragePath: zkeyStorageFilePath
+        }
+
+        // validate that the compiler hash is a valid hash 
+        const compiler = circuitData['compiler']
+        const compilerHashMatch = compiler['commitHash'].match(commitHashPattern)
+        if (compilerHashMatch.length === 0 || compilerHashMatch.length > 1) throw new Error("Error while setting up the ceremony. You should provide a valid commit hash of the circuit compiler.")
+
+        // validate that the verification options are valid
+        const verification = circuitData['verification']
+        if (verification['cfOrVM'] !== "CF" && verification['cfOrVM'] !== "VM") 
+            throw new Error("Error while setting up the ceremony. Please enter a valid verification mechanism: either CF or VM")
+
+        // @todo VM parameters verification
+        // if (verification['cfOrVM'] === "VM") {}
+
+        // check that the timeout is provided for the correct configuration
+        let dynamicThreshold: number | undefined
+        let fixedTimeWindow: number | undefined
+        if (data['timeoutMechanismType'] === CeremonyTimeoutType.DYNAMIC) {
+            if (circuitData['dynamicTreshold'] <= 0) 
+                throw new Error("Error while setting up the ceremony. The dynamic threshold should be > 0.")
+            dynamicThreshold = circuitData['dynamicTreshold']
+        }
+
+        if (data['timeoutMechanismType'] === CeremonyTimeoutType.FIXED) {
+            if (circuitData['fixedTimeWindow'] <= 0) 
+                throw new Error("Error while setting up the ceremony. The fixed time window threshold should be > 0.")
+            fixedTimeWindow = circuitData['fixedTimeWindow']
+        }
+
+        // the Circuit data for the ceremony setup
+        const circuit: CircuitDocument | CircuitInputData = {
+            name: circuitData['name'],
+            description: circuitData['description'],
+            prefix: circuitData['prefix'],
+            sequencePosition: i+1,
+            metadata: metadata,
+            files: files,
+            template: template,
+            compiler: compiler,
+            verification: verification,
+            fixedTimeWindow: fixedTimeWindow,
```

Comment:
nit: this will result in having one between `fixedTimeWindow` and `dynamicThreshold` set to null. Could this break things up?